pkg/catalogv2/git/download.go

```diff
@@ -1,124 +1,89 @@
 
-	return git.Head(branch)
-}
-
-func Update(secret *corev1.Secret, namespace, name, gitURL, branch string, insecureSkipTLS bool, caBundle []byte) (string, error) {
-	git, err := gitForRepo(secret, namespace, name, gitURL, insecureSkipTLS, caBundle)
+	err = r.hardReset("HEAD")
 	if err != nil {
 		return "", err
 	}
 
-	if isBundled(git) && settings.SystemCatalog.Get() == "bundled" {
-		return Head(secret, namespace, name, gitURL, branch, insecureSkipTLS, caBundle)
+	commit, err := r.getCurrentCommit()
+	return commit.String(), err
+}
+
+// CheckUpdate will check if rancher is in bundled mode,
+// if it is not in bundled mode, will make an update.
+// if it is in bundled mode, will just call Head method.
+func (r *repository) CheckUpdate(branch, systemCatalogMode string) (string, error) {
+	if isBundled(r.Directory) && systemCatalogMode == "bundled" {
+		return r.Head(branch)
 	}
 
-	commit, err := git.Update(branch)
-	if err != nil && isBundled(git) {
-		return Head(secret, namespace, name, gitURL, branch, insecureSkipTLS, caBundle)
+	commit, err := r.Update(branch)
+	if err != nil && isBundled(r.Directory) {
+		return r.Head(branch)
 	}
+
 	return commit, err
 }
 
-func Ensure(secret *corev1.Secret, namespace, name, gitURL, branch string, insecureSkipTLS bool, caBundle []byte) error {
-	git, err := gitForRepo(secret, namespace, name, gitURL, insecureSkipTLS, caBundle)
+// Update will check if repository exists, if exists will check for latest commit and update to it.
+// If the repository does not exist will try cloning again.
+func (r *repository) Update(branch string) (string, error) {
+	err := r.cloneOrOpen(branch)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return git.Ensure(branch)
-}
-
-func isBundled(git *git) bool {
-	return strings.HasPrefix(git.Directory, staticDir) || strings.HasPrefix(git.Directory, localDir)
-}
-
-func gitForRepo(secret *corev1.Secret, namespace, name, gitURL string, insecureSkipTLS bool, caBundle []byte) (*git, error) {
-	isGitSSH, err := isGitSSH(gitURL)
+	err = r.hardReset("HEAD")
 	if err != nil {
-		return nil, fmt.Errorf("failed to verify the type of URL %s: %w", gitURL, err)
-	}
-	if !isGitSSH {
-		u, err := url.Parse(gitURL)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse URL %s: %w", gitURL, err)
-		}
-		if u.Scheme != "http" && u.Scheme != "https" {
-			return nil, fmt.Errorf("invalid git URL scheme %s, only http(s) and git supported", u.Scheme)
-		}
+		return "", err
 	}
-	dir := gitDir(namespace, name, gitURL)
-	headers := map[string]string{}
-	if settings.InstallUUID.Get() != "" {
-		headers["X-Install-Uuid"] = settings.InstallUUID.Get()
+
+	commit, err := r.getCurrentCommit()
+	if err != nil {
+		return commit.String(), err
 	}
-	// convert caBundle to PEM format because git requires correct line breaks, header and footer.
-	if len(caBundle) > 0 {
-		caBundle = convertDERToPEM(caBundle)
-		insecureSkipTLS = false
+
+	lastCommit, err := r.getLastCommitHash(branch, commit)
```

Comment:
The `lastCommit` variable is just being used in a comparison with the `commit` variable, and I didn't see this function being used anywhere else. Could we maybe change it to something like `CheckForNewCommit` and just return a true/false?

Reply:
This way, any new future improvement to the code can just re-use the code.

Reply:
As we discussed, resolve the conversation please